### ISSUE TITLE
User avatars don't show in emails

### ISSUE THREAD

**Issue by ceorourke** [getsentry/sentry#58420]
At least for comment notifications, the avatar of the user who commented is just a blue box with a question mark regardless of whether they have a custom avatar or the default gravatar. We should check if this is happening for other notifications or if it's just the comment workflow email.

**Comment by ceorourke** [comment]
I went down a rabbithole looking into this and it's messed up in many ways. 

Firstly I learned about [this](https://docs.djangoproject.com/en/4.2/ref/models/instances/#django.db.models.Model.get_FOO_display) after being really confused that we have no function defined called `get_avatar_type_display()`.

Later on I found [this PR](https://github.com/getsentry/sentry/pull/58319) and looked into the usage of [format_html](https://docs.djangoproject.com/en/4.2/ref/utils/#django.utils.html.format_html). I printed out the value of it from [here](https://github.com/getsentry/sentry/blob/master/src/sentry/notifications/notifications/activity/base.py#L129) and it was only the value of the comment as simple text, all the HTML formatting that added the user avatar and everything built up in [description_as_html](https://github.com/getsentry/sentry/blob/master/src/sentry/notifications/notifications/activity/base.py#L186-L202) isn't there at all, because you're supposed to pass a format and then the variables that fit into it to `format_html`. 

The [generic template](https://github.com/getsentry/sentry/blob/master/src/sentry/templates/sentry/emails/activity/generic.html) renders `html_description` which should be what's returned by `description_as_html`, but then the [note template](https://github.com/getsentry/sentry/blob/master/src/sentry/templates/sentry/emails/activity/note.html) extends it and [note.html](https://github.com/getsentry/sentry/blob/master/src/sentry/templates/sentry/emails/activity/note.html#L17-L24) is duplicating the code written in `description_as_html`, and `author` doesn't seem to actually be passed to the template(I printed out the return value of [get_builder_args](https://github.com/getsentry/sentry/blob/master/src/sentry/mail/notifications.py#L149), so that never rendered anything). 

I also noticed that the `reason` (defined [here](https://github.com/getsentry/sentry/blob/master/src/sentry/notifications/types.py#L292-L304) and in the dict of data from `get_builder_args` mentioned above) is listed as "have changed the resolution status of this issue" for a comment when it should be "have commented on this issue", though I don't see that actually used anywhere.

tl;dr this is very messed up for workflow notifications that don't override the base class functions.
